src/main/java/org/jenkinsci/plugins/workflow/cps/GroovyShellDecorator.java

```diff
@@ -38,7 +38,26 @@ public void configureCompiler(@CheckForNull CpsFlowExecution context, CompilerCo
      */
     public void configureShell(@CheckForNull CpsFlowExecution context, GroovyShell shell) {}
 
+    /**
+     * Obtains a contextualized {@link GroovyShellDecorator} used to decorate the trusted shell.
+     *
+     * <p>
+     * By default, this method returns null decorator that doesn't do anything.
```

Comment:
Not obvious to me that this is a compatible change. In particular, did you try running `docker-workflow` tests against this?